safe/gui/tools/batch/batch_dialog.py

```diff
@@ -506,18 +510,25 @@ def run_task(self, task_item, status_item, count=0, index=''):
 
                         # generate map report and impact report
                         try:
-                            # this line is to save the impact report in default
-                            # InaSAFE directory.
-                            generate_impact_report(impact_function, self.iface)
-                            generate_impact_map_report(
-                                impact_function,
-                                self.iface)
+                            # generate report from component definition and
+                            # save it in the default output directory
+                            report_components = deepcopy(
+                                all_default_report_components)
+
+                            # don't generate infographic if exposure is not
+                            # a population
+                            exposure_type = layer_definition_type(
+                                impact_function.exposure)
+                            if exposure_type != exposure_population:
```

Comment:
Ok, I understand about the deep copy now. But can't we move this logic into the function? It's still code that need to be duplicated between dock, IFCW, batch runner ...